W-17455680 Einstein January update (#65)
mulesoft/docs-idp: modules/ROOT/pages/creating-document-actions.adoc

= Analyzing Documents Using Pre-Built Schemas

To analyze a generic document and fully customize the output fields and tables instead of using pre-built schemas, see xref:analyzing-documents-with-einstein.adoc[].

include::partial$document-action.adoc[tag=definition] 

include::partial$document-action.adoc[tag=details] 

> Create a document action using a predefined type as a template, and then specify which fields are mandatory, which fields to exclude from the JSON response, and the minimum confidence score expected for each field. 

Reviewer: ```suggestion
Create a document action using a predefined type as a template, and then specify the fields that are mandatory, the fields that must be excluded from the JSON response, and the minimum confidence score expected for each field. 
```

After you create a new document action, you can add reviewers and publish it to Anypoint Exchange, which enables RPA to execute the document action and also creates an API that you can call from any external system. 

When you create a document action, ensure that the configured schema works for each uploaded document. If you can't customize the schema to work with all sample documents, consider creating multiple document actions. For example, purchase orders from multiple vendors might be different enough to require a separate document action for each vendor.

Creating a document action with a pre-built schema requires the following tasks:

. <<upload-files>>
. <<configure-schema>> 
+

> Unless you select the Generic document action, which does not use a schema.

Reviewer: ```suggestion
Unless you select the Generic document action, which doesn't use a schema.
```
